chessington/engine/pieces.py

```diff
@@ -187,5 +213,42 @@ class King(Piece):
                   Square(1, 1), Square(-1, 1), Square(-1, -1), Square(1, -1)]
     NOT_KING_OR_KNIGHT = 0
 
+    def is_attacked(self, board, current_square):
+        # Needs modularization
+        for row in range(0, 8):
+            for col in range(0, 8):
+                selected_square = Square.at(row, col)
+                selected_piece = board.get_piece(selected_square)
+                if selected_piece and selected_piece.player != self.player:
+                    if type(selected_piece) is King:
+                        if abs(selected_square.row - current_square.row) <= 1 and abs(
+                                selected_square.col - current_square.col) <= 1:
```

Comment:
Wouldn't this need to be `==1`? If it would be `<1`, wouldn't there be a case where the King is next to the opposite King?